src/webgl-utils/track-context-state.js

```diff
@@ -261,27 +255,39 @@ class GLState {
 
   pop() {
     assert(this.stateStack.length > 0);
-    const oldValues = this.stateStack.pop();
+    // Use the saved values in the state stack to restore parameters
+    const oldValues = this.stateStack[this.stateStack.length - 1];
```

Comment:
I see you found it, this is the reason for some of layers not working with my deck.gl diff.